### PR TITLE
Feature/#25/aggressive mobs

### PR DESCRIPTION
**Pull request by ivoryfitz**
Got some clean-up to do still, but I figure I'd go ahead and get it pushed up here since it _seems_ like it's working. There are definitely more abstractions that could be made to JEnemyBody2D as we are duplicating a fair amount of code from the Sheep and TreeTrunkGuy, but I figure we can go by the rule of three and clean that up when we do the Molded Druvar.

Anyways, I need to fix a bunch of naming also, I'm still a little confused on Scene naming vs scripts. Camel case vs. snake case vs. hyphenating type thing. So if anyone has guidance on that one holler.

There's....a lot in this PR...sorry in advance reviewers 😅

### CODE REVIEW

**Comment by jonathaneeckhout** [comment]
The convention that I was using for the naming was:

Scenes are always Camel case, same goes for the scripts attached to the scene.

Classes scripts should have Camel case as well.

Scripts without any class are just all lower case.

I know it does not make that much sense for the scripts without any class so I am up to change all file names to be camel case.

File: scenes/enemies/Sheep/Sheep.tscn [hunk 1]
```diff
@@ -210,28 +212,36 @@ libraries = {
...
+
 [node name="BeehaveTree" type="Node" parent="." node_paths=PackedStringArray("blackboard")]
 script = ExtResource("5_yme22")
 enabled = false
 actor_node_path = NodePath("..")
 blackboard = NodePath("../Blackboard")
 
-[node name="SelectorComposite" type="Node" parent="BeehaveTree"]
-script = ExtResource("6_5ksd8")
+[node name="What am I doing with my life?" type="Node" parent="BeehaveTree"]
```
Comment (jonathaneeckhout) [inline scenes/enemies/Sheep/Sheep.tscn:230]: ROFL with this name.

Anyway I think it's better to remove the spaces and I guess the question mark as well.

WhatAmIDoingWithMyLife

**Comment by NancokPS2** [comment]
> The convention that I was using for the naming was:
> 
> Scenes are always Camel case, same goes for the scripts attached to the scene.
> 
> Classes scripts should have Camel case as well.
> 
> Scripts without any class are just all lower case.
> 
> I know it does not make that much sense for the scripts without any class so I am up to change all file names to be camel case.

That's a pretty good idea, being able to tell which is a ClassDB class and which is a generic script sounds very helpful.

**Comment by jonathaneeckhout** [comment]
When testing it, I can get it to crash consistently:
Screenshot from 2023-10-12 13-28-47

Reproduction scenario:
I bough an axe and the leather set from vendor (not sure if this is important).
Walked towards a treetrunkguy and let it aggro me.

I did notice that it did not occur every time, but like 50% of the times I do these steps.

**Comment by jonathaneeckhout** [comment]
@ivoryfitz  could you make it so that the treetrunkguys actually hit the player? This step is needed to test the respawn mechanism in a real usecase.

**Comment by ivoryfitz** [comment]
@jonathaneeckhout Will look at the crash, not sure what's up with that.

They should be hitting the player. Currently using `treetrunkguy.attack(target)` to do that. Is there a different way to actually trigger attacks that I should be using?

**Comment by ivoryfitz** [comment]
> The convention that I was using for the naming was:
> 
> Scenes are always Camel case, same goes for the scripts attached to the scene.
> 
> Classes scripts should have Camel case as well.
> 
> Scripts without any class are just all lower case.
> 
> I know it does not make that much sense for the scripts without any class so I am up to change all file names to be camel case.

I'm fine with just moving forward like that and if the naming starts to not make sense later we can change it. For now, just want to match the style for consistency sake so I'll change to match what outline here.

**Comment by jonathaneeckhout** [comment]
> @jonathaneeckhout Will look at the crash, not sure what's up with that.
> 
> They should be hitting the player. Currently using `treetrunkguy.attack(target)` to do that. Is there a different way to actually trigger attacks that I should be using?

I don't think so, I just didn't see any of the hp of the player being removed when they were hitting. Could be that the crash was preventing them to reach that stage.

I expect that the crash will be timing related as you don't see it and I do.

**Comment by ivoryfitz** [comment]
Yea I think I found it. It is reproducible on my end right now.

**Comment by ivoryfitz** [comment]
I'm unsure on the issue with the player not getting attacked/the health bar updating.

The attacks are getting issued here: https://github.com/jonathaneeckhout/jdungeon/pull/63/files#diff-5e16fcaa89700f20a9bac47817dddc38b3bccf8d5deff6f3d4c6190e11afb6aaR18

And I verified with logs that the "hurt" function is getting triggered for the player. 

I'm thinking it can be a separate issue? I can look into it more, but I think if its an existing issue, I don't want to make this PR any bigger than it already is 😅

**Comment by jonathaneeckhout** [comment]
Great work, this is a big one.

I did find 2 issues (1 is open for discussion) but like you said, it's better to tackle them in a separate PR so I made new issues for them:
https://github.com/jonathaneeckhout/jdungeon/issues/68
https://github.com/jonathaneeckhout/jdungeon/issues/69

**Pull request merged**